### PR TITLE
Adds ability to ban email by partial match on disposable email service list

### PR DESCRIPTION
**Pull request by ritec**
- Adds the ability to block against disposable email services that use subdomains by doing a partial match against the email domain. 

For example, if we add `tempmail.com` to the `disposable_email_services` list in the `valid_email.yml` file, we can now block emails that use `fake@sub.tempmail.com` or `fake@another.sub.tempmail.com` as well. All we have to do is use the `ban_partial_disposable_email` configuration option in our model validation. 

The way this works is not by finding a perfect match between domains but by checking if the email domain ends with a string from the `disposable_email_services` list.

- Also fixes some specs that were not passing due to a typo using `msgs` instead of `msg` in an earlier release.

CC: For your consideration @hallelujah

### CODE REVIEW

File: lib/valid_email/email_validator.rb [hunk 1]
```diff
@@ -20,9 +20,14 @@ def validate_each(record,attribute,value)
...
       r = BanDisposableEmailValidator.new(:attributes => attributes, message: options[:message]).validate(record)
     end
+    # Check if part of domain is in disposable list
+    if r && options[:ban_partial_disposable_email]
+      require 'valid_email/ban_partial_disposable_email_validator'
+      r = BanPartialDisposableEmailValidator.new(:attributes => attributes, message: options[:message]).validate(record)
+    end
     unless r
       msg = (options[:message] || I18n.t(:invalid, :scope => "valid_email.validations.email"))
-      record.errors.add attribute, message: I18n.interpolate(msgs, value: value)
```
Comment (hallelujah) [inline lib/valid_email/email_validator.rb:25]: Good catch! There was a missing test here.

File: lib/valid_email/ban_partial_disposable_email_validator.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  def self.config=(options)
+    @@config = options
+  end
+
+  # Required to use config outside
+  def self.config
+    @@config = [] unless defined? @@config
+
+    @@config
+  end
```
Comment (hallelujah) [inline lib/valid_email/ban_partial_disposable_email_validator.rb:15]: Why is the configuration inside the validator? I think it should go to a configuration for the overall ValidEmail gem

File: lib/valid_email/email_validator.rb [hunk 3]
```diff
@@ -20,9 +20,14 @@ def validate_each(record,attribute,value)
       require 'valid_email/ban_disposable_email_validator'
       r = BanDisposableEmailValidator.new(:attributes => attributes, message: options[:message]).validate(record)
     end
+    # Check if part of domain is in disposable list
+    if r && options[:ban_partial_disposable_email]
+      require 'valid_email/ban_partial_disposable_email_validator'
+      r = BanPartialDisposableEmailValidator.new(:attributes => attributes, message: options[:message]).validate(record)
+    end
```
Comment (hallelujah) [inline lib/valid_email/email_validator.rb:27]: Can it be moved to `BanDisposableEmailValidator` with options `partial: true`

**Review by hallelujah — COMMENTED** [review]
Hello @ritec! Thank you for your contribution ❤️ 

I think it would be better to integrate this change to `BanDisposableEmailValidator` with an option `partial: true`. What do you think?

**Comment by ritec** [comment]
Hey thanks for reviewing, I will work on this ASAP so we can try to merge it in.

File: lib/valid_email/email_validator.rb (hunk 3 above)
Comment (ritec) [reply lib/valid_email/email_validator.rb]: Done!

**Comment by ritec** [comment]
@hallelujah I have implemented the reviews. Let me know if it is looking good and if so I will squash so it can be merged.

**Review by hallelujah — APPROVED** [review]
LGTM thank you!

**Comment by zachmargolis** [comment]
Hi there, we'd love to use this functionality in our app, can we get new release of the gem that includes this?

**Comment by ritec** [comment]
@zachmargolis Unfortunately the gem version was not bumped when this was merged. @hallelujah , can you create a new release?

**Pull request merged**